drivers/adc/adc_mcux_lpadc.c

```diff
@@ -225,6 +225,16 @@ static int mcux_lpadc_channel_setup(const struct device *dev,
 		}
 	} else if (channel_cfg->reference == ADC_REF_EXTERNAL0) {
 		LOG_DBG("ref external0");
+# if defined(FSL_FEATURE_LPADC_HAS_INTERNAL_TEMP_SENSOR)
+	} else if (channel_cfg->reference == ADC_REF_INTERNAL) {
```

Comment:
nit: use `IS_ENABLED(FSL_FEATURE_LPADC_HAS_INTERNAL_TEMP_SENSOR)` instead of `#if`